packages/x-tailwindcss/src/x-tailwind-plugin/components/product-image/index.ts

```diff
@@ -0,0 +1,30 @@
+import { rename } from '@empathyco/x-utils';
+import { deepMerge } from '@empathyco/x-deep-merge';
+import { TailwindHelpers } from '../../../types';
+import { productImageDefault } from './default';
+import { overlay } from './overlay';
+import { zoom } from './zoom';
+
+/**
+ * Returns the component `input` CSS.
+ *
+ * @param helpers - The {@link TailwindHelpers} to generate CSS.
+ * @returns The {@link CssStyleOptions} for the component.
+ */
+// eslint-disable-next-line  @typescript-eslint/explicit-function-return-type
+export function productImage(helpers: TailwindHelpers) {
+  return {
+    '.product-image': deepMerge(
+      productImageDefault(helpers),
+      rename(
+        {
+          ...zoom(),
+          ...overlay()
+        },
+        {
+          prefix: '&-'
+        }
+      )
+    )
+  };
```

Comment:
```suggestion
  return {
    '.product-image': {
      ...productImageDefault(helpers),
      ...rename(
        {
          ...zoom(),
          ...overlay()
        },
        {
          prefix: '&-'
        }
      )
    }
  };
```
I think we can skip the `deepMerge` since we don't want to merge nested objects